pkg/controller/persistentvolume/persistentvolume_controller.go

```diff
@@ -641,25 +641,39 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *api.PersistentVolume)
 			return nil
 		} else {
 			// Volume is bound to a claim, but the claim is bound elsewhere
-			if hasAnnotation(volume.ObjectMeta, annBoundByController) {
-				// This is part of the normal operation of the controller; the
-				// controller tried to use this volume for a claim but the claim
-				// was fulfilled by another volume. We did this; fix it.
-				glog.V(4).Infof("synchronizing PersistentVolume[%s]: volume is bound by controller to a claim that is bound to another volume, unbinding", volume.Name)
-				if err = ctrl.unbindVolume(volume); err != nil {
+			if hasAnnotation(volume.ObjectMeta, annDynamicallyProvisioned) && volume.Spec.PersistentVolumeReclaimPolicy == api.PersistentVolumeReclaimDelete {
+				// This volume was dynamically provisioned for this claim. The
+				// claim got bound elsewhere, and thus this volume is not
+				// needed. Delete it.
```

Comment:
Comment on the fact that, if you set it to a reclaim policy other than delete, this might leak volumes?  We expect dynamic provisioning to imply dynamic deleteion, but it's not necessarily so...